universe/Fleet.cpp

```diff
@@ -1156,6 +1156,24 @@ void Fleet::CalculateRouteTo(int target_system_id) {
     }
 }
 
+bool Fleet::Blockaded() const {
+    /** returns true if fleet is in a system and is prevented from using at least
+        one starlane entry to leave its current system */
+    std::shared_ptr<System> system = GetSystem(this->SystemID());
+
+    if (!system)
+        return false;
+
+    const std::map<int, bool> adjacent_system_IDs = system->VisibleStarlanesWormholes(this->Owner());
```

Comment:
Thank you, that fixed it, and also thank you that you took the time. I will update the PR soon.